### DISCUSSION TITLE
Staging-server for portalen

### DISCUSSION THREAD

**Discussion by piofinn** [fremtind/jokul#2278] [Annet]
Det hadde vært fint å ha en staging-server for portalen, slik at det er lett å vise frem kjørende kode når man lager en ny branch/PR. Dette er særlig nyttig for designerne, som gjerne ikke har utviklingsmiljø på maskinene sine.

Det er nok begrenset/ingen mulighet for dette med GitHub Pages, men man kan se på andre alternativer som Zeit Now, Heroku, eller liknende. Alt vi trenger er et byggsteg som bygger portalen og legger resulterende filer på en statisk host 😊

**Comment by eduardodir** [comment]
Experimenterar med Zeit

**Comment by joms** [comment]
Fikk Zeit sin GitHub integrasjon til å fungere på følgende måte:

la til now.json i prosjektrot.
```json
{
    "version": 2,
    "rewrites": [{
            "source": "/jokul/(.*)",
            "destination": "/$1"
        },
        {
            "source": "/(.*)",
            "destination": "/index.html"
        }
    ]
}
```

la til config i now.sh

![image](https://user-images.githubusercontent.com/537060/79549733-a5e6ba00-8097-11ea-9264-53a3fba91238.png)

Kjenner likevel at jeg ikke er helt fan av integrasjonen, og tror vi kan vinne mer på å gjøre dette med now-cli i GitHub Actions.

Merk: Jeg testet dette på en fork for ikke å release masse greier ved et uhell :D

**Comment by eduardodir** [comment]
Syns du borde även göra en issue om detta å "förbättra" dokumentationen :)
Om vi kan ha detta så länge, så kan vi se vidare på now-cli i GitHub Actions efterpå.
now.json borde ju kunnat vart helt konfiguerarbar å ikke en del i webui som vi pratat om, kanske en issue till om det till dem?

**Comment by wkillerud** [comment]
Jeg bruker [Netlify](https://www.netlify.com) til den personlige siten min og har vært fornøyd med opplevelsen der med previews for pull requests. De har noe [dokumentasjon rundt bruk i monorepo](https://docs.netlify.com/configure-builds/common-configurations/monorepos/).

**Comment by joms** [comment]
Vi har vært inne på tanken med Netlify eller Vercel også, men det er et poeng å prøve å bruke AWS siden det er det som skal være hovedplattformen vår på sikt.

**Comment by onordbo** [comment]
+1 på en staging server med noen preview-lenker på PR'ene. Det hadde gjort livet veldig, veldig mye enklere for oss som skal reviewe visuelle endringer i portalen 😇

**Comment by wkillerud** [comment]
- #2701

![It's alive!](https://media.giphy.com/media/11RwocOdukxqN2/giphy-downsized-large.gif)
